### ISSUE TITLE
[SkipToContent]: no documentation, fails DAP contrast, unnecessary required prop

### ISSUE THREAD

**Issue by vpicone** [carbon-design-system/carbon#4802]
The `SkipToContent` component and it's functionality are critical to accessible apps, we should definitely be polishing this.

Issues with the `SkipToContent` component:
- this component's existance/props are effectively undocumented (even in stories where it is used) storybook isn't picking up many of individual the shell components
- the component fails DAP contrast static analysis. DAP is comparing the default link color to the shell background. This could be fixed by making the link white at all times, not just during focus.
- The component **requires** an href but provides one as a default. Even in our stories, we don't provide one. Allowing use of default is fine, we shouldn't throw prop-type warnings when folks do so.

1. Navigate to [a story using `SkipToContent`]( https://react.carbondesignsystem.com/?path=/story/ui-shell--header-base-w-navigation-and-actions)
2. Click show info. Notice no prop table or story source
3. Run DAP
